Convert deeplink to global queue & add error dialog to video player

1hitsong opened the pull request:
<!-- markdownlint-disable MD041 first-line-heading -->
## Changes
Converts deep link content to use new global queue. This also adds the error dialog to a few more existing valid checks to tell the user the video is unplayable.

cewert reviewed (changes requested):
There's a regression when using the launch command. Currently unstable works when testing for both launch and input commands but this PR only works for input commands. To reproduce:

- launch this PR to the roku app and login with demo server then exit the channel and debugger
- go to demo web client and grab a videos id from the URL i.e. grab `5e6e8380563c5211106652362c5c6843` from `index.html#!/details?id=5e6e8380563c5211106652362c5c6843&serverId=713dc3fe952b438fa70ed35e4ef0525a`
-  open the [roku deep linking tester](http://devtools.web.roku.com/DeepLinkingTester/) (I'm using pc version)
- select the jellyfin dev channel at the bottom
- create new test using id from second step like the picture below
![deep-link-testing](https://github.com/jellyfin/jellyfin-roku/assets/16514652/94170871-9a7c-4c1d-bd9f-1c473249df5a)
- use the dropdown and then select `Send (Launch)` button
- observe

1hitsong commented:
I followed your instructions and it launched the client and autoplayed Night of the Living Dead. See video:

https://github.com/jellyfin/jellyfin-roku/assets/3330318/ca5ad39a-e18d-451e-be62-795eff1cd106

I also checked it on my local server by launching a valid movie ID and an invalid movie ID. See video:

https://github.com/jellyfin/jellyfin-roku/assets/3330318/26726630-d733-4522-aa74-16aecd6199c3

What issue were you seeing (or not seeing) because as you see in the video, these examples work as I would expect them to.

cewert commented:
I just tested again and I'm still getting the "Error during playback:  An error was encountered while playing this item" popup but only for the launch command. When I test the unstable branch it looks like your first video.

cewert commented on components/video/VideoPlayerView.brs:
```diff
@@ -64,23 +64,48 @@ sub onSubtitleChange()
         return
     end if
 
     if m.LoadMetaDataTask.content.count() = 0
```
This check is a duplicate of the one above it. `content.count()`) has to be greater than 0 for `content[0]` to be valid

cewert commented on source/Main.brs:
```diff
@@ -82,19 +82,14 @@ sub Main (args as dynamic) as void
-            dialog = createObject("roSGNode", "Dialog")
-            dialog.id = "OKDialog"
-            dialog.title = tr("Not found")
-            dialog.message = tr("The requested content does not exist on the server")
```
When testing unstable with an invalid video id (both launch and input) we receive this error message but now we just get the generic error message - "Error during playback: An error was encountered while playing this item". Can we restore the old message or something similar to it when the movie id is invalid?

Also, the translation file will need to be updated if we are no longer using these popup messages or we decide to change them.

cewert replied:
When I tested the Input command with an invalid `contentId` I got this error message "The requested content does not exist on the server" but when using Launch I got "There was an error retrieving the data for this item from the server."

1hitsong replied:
Couple of things to unpack here.

**Why the change in the error message text?**
The original error message "The requested content does not exist on the server" makes a major assumption about why the video component didn't load any content. If **_any_** error occurs when retrieving the video content, it responds with this message; however, it's assuming the error reason, there is no explicit "does this exist on the server" check occurring. That's why I changed it to be accurate based on what we actually know.

**The different message from the input command**
The input command functionality is handled in a different block of code that deals with input events. It's not part of the code labeled as deep link code. It will be adjusted as part of a different PR.

![image](https://github.com/jellyfin/jellyfin-roku/assets/3330318/c93254d5-82cb-4ddc-8dc9-51edf1da1ed1)

cewert commented on components/video/VideoPlayerView.brs:
```diff
@@ -275,12 +291,7 @@ sub bufferCheck(msg)
-            dialog = createObject("roSGNode", "Dialog")
-            dialog.title = tr("Error Retrieving Content")
-            dialog.buttons = [tr("OK")]
-            dialog.message = tr("There was an error retrieving the data for this item from the server.")
```
Although this error message here wasn't super useful on it's own, it would allow someone to figure out where in the code the error message was coming from vs now where it has the same error message as other places in the code.

Can we modify `showPlaybackErrorDialog()` to accept a `message` param? That way we could keep this error message.

1hitsong commented:
Updated the error function to accept a message string. I also updated all calls to have error messages that are specific to the encountered error.

All error messages are pre-existing and did not require new translations.
